sn_auditor/src/dag_db.rs

```diff
@@ -341,17 +412,87 @@ impl SpendDagDb {
         beta_tracking.total_accumulated_utxo += spend.spend.spent_tx.outputs.len() as u64;
         beta_tracking.total_on_track_utxo += utxos_for_further_track;
 
+        // Collect royalties
+        let royalty_pubkeys: BTreeSet<_> = spend
+            .spend
+            .network_royalties
+            .iter()
+            .map(|derivation_idx| NETWORK_ROYALTIES_PK.new_unique_pubkey(derivation_idx))
+            .collect();
+        let default_royalty_pubkeys: BTreeSet<_> = spend
+            .spend
+            .network_royalties
+            .iter()
+            .map(|derivation_idx| DEFAULT_NETWORK_ROYALTIES_PK.new_unique_pubkey(derivation_idx))
+            .collect();
+        let mut royalties = BTreeMap::new();
+        for output in spend.spend.spent_tx.outputs.iter() {
+            if default_royalty_pubkeys.contains(&output.unique_pubkey)
+                || royalty_pubkeys.contains(&output.unique_pubkey)
+            {
+                let _ = royalties.insert(
+                    SpendAddress::from_unique_pubkey(&output.unique_pubkey),
+                    output.amount.as_nano(),
+                );
+            }
+        }
+
+        if royalties.len() > (spend.spend.spent_tx.outputs.len() - 1) / 2 {
```

Comment:
What if there is no change output because by chance the amount is perfect?
In this case the `-1` is unnecessary. Although there is no way to know which of the outputs is the change...

Reply:
this is just for alerting suspicious spend.
with the main nano supplier is currently controlled by us, the case of no change is really rare.
so, shall not be a problem here.